src/Materials/Textures/babylon.texture.ts

```diff
@@ -117,6 +117,8 @@
 
             scene = this.getScene();
 
+            scene.getEngine().onBeforeTextureInitObservable.notifyObservers(this);

```

Comment:
we need to make sure that this.url is used later on by the constructor (and not just url)